File: server-data/resources/[ui]/react_cad/client/markers.lua
Citizen.CreateThread(function()

	for k, zone in pairs(Config.Zones) do
		zone._marker = MarkerClass:new({
			type = 1,
			pos = zone.Position,
			color = { r = 55, g = 100, b = 255 },
			scale = { x = 2, y = 2, z = 1},
			drawDistance = zone.DrawDistance or 20,
			onPress = 'react_cad:onPress',
			onEnter = 'react_cad:onEnter',
			onExit = 'react_cad:onExit',
--			notification = 'Нажмите ~INPUT_PICKUP~ для доступа к шкафчику',
		})

		zone._sprite = Sprite3DClass:new({
			pos = zone.Position,
			scale = { x = 0.1, y = 0.1 },
			textureDict = 'spritedict',
			textureName = 'craft',
			alpha = 255,
			color = { r = 255, g = 255, b = 255 },
			drawDistance = 5.0,
			distanceFade = false,
		})
		zone.id = k

		zone._marker:set('Zone', zone)
	end

	while true do
		Citizen.Wait(0)
		local coords = GetEntityCoords(PlayerPedId(-1))
		for _, zone in pairs(Config.Zones) do
			zone._marker:draw(coords)
			zone._sprite:draw(coords)
		end
	end
end)

RegisterNetEvent("react_cad:onPress")
AddEventHandler("react_cad:onPress", function(marker)
	local zone = Config.Zones[marker.Zone.id]
	TriggerServerEvent('react_cad:menu_show', zone.HomePage, zone.Permissions)
end)

RegisterNetEvent("react_cad:onEnter")
AddEventHandler("react_cad:onEnter", function(marker)
	Config.Zones[marker.Zone.id]._sprite:set('textureName', 'craft_e')
end)

RegisterNetEvent("react_cad:onExit")
AddEventHandler("react_cad:onExit", function(marker)
	Config.Zones[marker.Zone.id]._sprite:set('textureName', 'craft')
end)

File: server-data/resources/EngineTest/src/client/atm/markers.ts
/* eslint-disable no-param-reassign */
import { Marker, Sprite, Blip } from '@nova/engine-lib/client/Graphics';
import { Delay, Vector2, Vector3 } from '@nova/engine-lib/shared';

import Config from '../../config/atm';
import State from './State';

import { showBank, hideBank } from './ui';

// Configure markers, sprites, blips
Config.Zones.forEach((zone, index) => {
  zone.sprite = new Sprite({
    coords: zone.Position,
    scale: new Vector2(0.1, 0.1),
    textureDict: 'spritedict',
    textureName: 'locker',
    alpha: 255,
    color: [255, 255, 255],
    drawDistance: 5.0,
    // distanceFade: false,
  });

  zone.marker = new Marker({
    markerType: 1,
    coords: zone.Position,
    color: [55, 100, 255],
    scale: Vector3.FromArray([2, 2, 1]),
    drawDistance: zone.DrawDistance || 20.0,
    notificationText: 'Нажмите ~INPUT_PICKUP~ для доступа к банку',

    onPress: () => {
      showBank(zone);
    },
    onEnter: () => {
      // zone.sprite.textureName = 'locker_e';
      State.ATMEntity = null;
      State.BankZone = zone;
    },
    onExit: () => {
      // zone.sprite.textureName = 'locker';
      hideBank();
      // ESX.SetPlayerData('inventory_current_container', nil)
    },
  });

  const blip = new Blip({
    coords: zone.Position,
    sprite: 108,
    display: 4,
    scale: 0.7,
    color: 30,
    shortRange: true,
    text: 'Банк',
  });

  zone.blip = blip;
  zone.id = index;
});

// Draw
Delay(1000).then(() =>
  setTick(async () => {
    Config.Zones.forEach((zone: { marker: Marker; sprite: Sprite }) => {
      zone.marker.draw();
      zone.sprite.draw();
    });
  }),
);
